docs(manifest): Improvements to `name`, `short_name`, `start_url` (#35847)
mdn/content: files/en-us/web/manifest/name/index.md

{{QuickLinksWithSubpages("/en-US/docs/Web/Manifest")}}

> The `name` manifest member is used to specify the name of your web application as it's typically displayed to users.

Reviewer: Loss of information
```suggestion
The `name` manifest member is used to specify the name of your web application as it's usually displayed to users, such as in application lists or as a label for an icon.
```

Author: Sure, can add. Tweaked the last bit to "or as a label for your application's icon."

## Syntax

```json-nolint
/* Full names of web apps */
"name": "Daily Task Planner"

> "name": "Recipe and Pantry Tracker"

Reviewer: Let's cut back to two (no real strong reason, other than I think the point has been made).
```suggestion
```

"name": "Mindful Meditation"
```

### Values

- `name`
  - : A string that specifies the full name of your web app.

## Description

The `name` manifest member serves as the {{Glossary("Accessible_name", "accessible name")}} for your installed app. It is displayed to users in various contexts, such as in a list of other installed web apps, as a label for your app's icon, or in the application switcher or task manager. In space-constrained contexts, such as on a device's home screen or in the application switcher, where the full `name` may not fit, the [`short_name`](/en-US/docs/Web/Manifest/short_name) value may be used instead, if available.

> [!NOTE]
> For Chromium-based browsers, a `name` or `short_name` is required in the manifest for the web app to be installable.

> Consider the following points when selecting a name for your web app:

Reviewer: Worth having a best practices heading? Just a thought, not advocating.

Author: Generally not a fan of a lone H3 (as an exception, we do have them in "Examples" section), but adding one here will improve the structure and readability of the information.

> - How well it indicates the purpose or nature of your app
> - Whether it is clear and easy to understand and remember
> - Its length, especially if you're not providing a separate `short_name`
> - How it will appear in different contexts, such as in app lists or home screen
> - Its uniqueness compared to existing apps
> - Its cultural sensitivity and appropriateness for your target audience
> - How well it translates or is perceived in different languages if your app targets a global audience
> - Whether it has any trademark conflicts

Reviewer: Might be more readable to make these succinct

```suggestion
- How well it indicates the purpose or nature of your app
- Whether it is clear and easy to understand and remember
- Length of the name, especially if you're not providing a separate `short_name`
- How it will appear in different contexts, such as in app lists or home screen
- How easily it can be differentiated from other similar apps
- Cultural sensitivity and appropriateness for your target audience
- How well it translates or is perceived in different languages if your app target is a global audience
- Potential trademark conflicts
```

Author: These are good, incorporating. Also changing the order slightly so that the length consideration is stated upfront:

```
- Length of the name, especially if you're not providing a separate `short_name`
- How well it indicates the purpose or nature of your app
- Whether it is clear and easy to understand and remember
- How it appears in different contexts, such as in app lists or home screen
- How easily it can be differentiated from other similar apps
- Cultural sensitivity and appropriateness for your target audience
- How well it translates or is perceived in different languages if your app target is a global audience
- Potential trademark conflicts
```

## Examples

### Adding a name for your web app

> For a web app that helps users navigate trails and plan their hiking adventures, you can add the following `name` to the app manifest:

Reviewer: Since this is "for example":

```suggestion
For a web app that helps users navigate trails and plan their hiking adventures, you might add the following `name` to the app manifest:
```

{{QuickLinksWithSubpages("/en-US/docs/Web/Manifest")}}

The `name` manifest member is used to specify the full name of your web application as it's usually displayed to users, such as in application lists or as a label for your application's icon.

## Syntax

```json-nolint
/* Full names of web apps */
"name": "Daily Task Planner"
"name": "Recipe and Pantry Tracker"
```

### Values

- `name`
  - : A string that specifies the full name of your web app.

## Description

> The `name` manifest member serves as the {{Glossary("Accessible_name", "accessible name")}} for your installed app. It is displayed to users in various contexts, such as in a list of other installed web apps, as a label for your app's icon, or in the application switcher or task manager. In space-constrained contexts, such as on a device's home screen or in the application switcher, where the full `name` may not fit, the [`short_name`](/en-US/docs/Web/Manifest/short_name) value may be used instead, if available.

Reviewer: The short name is a different "context" so let's put in own paragraph.

```suggestion
The `name` manifest member serves as the {{Glossary("Accessible_name", "accessible name")}} for your installed app. It is displayed to users in various contexts, such as in a list of other installed web apps, as a label for your app's icon, or in the application switcher or task manager.

In space-constrained contexts, such as on a device's home screen or in the application switcher, where the full `name` may not fit, the [`short_name`](/en-US/docs/Web/Manifest/short_name) value may be used instead, if available.
```

{{QuickLinksWithSubpages("/en-US/docs/Web/Manifest")}}

The `name` manifest member is used to specify the full name of your web application as it's usually displayed to users, such as in application lists or as a label for your application's icon.

## Syntax

```json-nolint
/* Full names of web apps */
"name": "Daily Task Planner"
"name": "Recipe and Pantry Tracker"
```

### Values

- `name`
  - : A string that specifies the full name of your web app.

## Description

The `name` manifest member serves as the {{Glossary("Accessible_name", "accessible name")}} for your installed app. It is displayed to users in various contexts, such as in a list of other installed web apps, as a label for your app's icon, or in the application switcher or task manager.

In space-constrained contexts where the full `name` may not fit, such as on a device's home screen or in the application switcher, the value of the [`short_name`](/en-US/docs/Web/Manifest/short_name) property (if defined) may be used instead.

> [!NOTE]

> > For Chromium-based browsers, a `name` or `short_name` is required in the manifest for the web app to be installable.

Reviewer: Are you planning on removing these before merge? Please link the BCD PR.

Author: Done here: https://github.com/mdn/browser-compat-data/pull/24700
